### ISSUE TITLE
parsable commands.json

### ISSUE THREAD

**Issue by edge226** [antirez/redis-doc#834]
I'm writing a bash library for redis and as a quick fix I created an array of groups, I then wrapped the array in a for loop and used `redis-cli help @groupname` to parse out lines containing all command names.

The moment I went to pass help information to the server through a resp string I got the error `-ERR unknown command 'help'` in return.

I continued my search for a solution and found commands.json which has *ALL* the information I need but I cannot manage to get the information extracted with jq because its got no field. There is no `.[commands]` for jq or anything like that I can use.

Is there any easy way I can get a simple commands listing with resp or from commands.json?

Am I incorrect to think that there is no "Field" this data is not parsable and that commands.json is mean to parse information *about* commands but not the name field?

**Comment by itamarhaber** [comment]
Hello Edward,

The `COMMAND` command can probably help you with that - see
https://redis.io/commands/command

Cheers,
Itamar


On May 30, 2017 8:30 PM, "Edward Tunnah" <notifications@github.com> wrote:

I'm writing a bash library for redis and as a quick fix I created an array
of groups, I then wrapped the array in a for loop and used redis-cli help
@groupname to parse out lines containing all command names.

The moment I went to pass help information to the server through a resp
string I got the error -ERR unknown command 'help' in return.

I continued my search for a solution and found commands.json which has *ALL*
the information I need but I cannot manage to get the information extracted
with jq because its got no field. There is no .[commands] for jq or
anything like that I can use.

Is there any easy way I can get a simple commands listing with resp or from
commands.json?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/antirez/redis-doc/issues/834>, or mute the thread
